src/microbe_stage/editor/MicrobeEditorActionData.cs

```diff
@@ -0,0 +1,333 @@
+﻿using System;
+using System.Collections.Generic;
+
+/// <summary>
+///   Describes how two microbe actions interference with each other.
+/// </summary>
+/// <para>Used for MP calculation</para>
+public enum MicrobeActionInterferenceMode
+{
+    /// <summary>
+    ///   The two actions are completely independent
+    /// </summary>
+    NoInterference,
+
+    /// <summary>
+    ///   The other action replaces the this one
+    /// </summary>
+    ReplacesOther,
+
+    /// <summary>
+    ///   The two actions cancel out each other
+    /// </summary>
+    CancelsOut,
+
+    /// <summary>
+    ///   The two actions can be combined to a whole different action.
+    ///   Call <see cref="MicrobeEditorActionData.Combine"/> to get this action.
+    /// </summary>
+    Combinable,
+}
+
+/// <summary>
+///   This is its own interface to make JSON loading dynamic type more strict
+/// </summary>
+public abstract class MicrobeEditorActionData
+{
+    /// <summary>
+    ///   Does this action cancel out with the <paramref name="other"/> action?
+    /// </summary>
+    /// <returns>
+    ///   Returns the interference mode with <paramref name="other"/>
+    /// </returns>
+    /// <para>Do not call with itself</para>
+    public abstract MicrobeActionInterferenceMode GetInterferenceModeWith(MicrobeEditorActionData other);
+
+    /// <summary>
+    ///   Combines two actions to one if possible.
+    ///   Call <see cref="MicrobeEditorActionData.GetInterferenceModeWith"/> first and check if it returns <see cref="MicrobeActionInterferenceMode.Combinable"/>
+    /// </summary>
+    /// <param name="other">The action this should be combined with</param>
+    /// <returns>Returns the combined action</returns>
+    /// <exception cref="NotSupportedException">Thrown when combination is not possible</exception>
+    public MicrobeEditorActionData Combine(MicrobeEditorActionData other)
+    {
+        if (GetInterferenceModeWith(other) != MicrobeActionInterferenceMode.Combinable)
+            throw new NotSupportedException();
+
+        return CombineGuaranteed(other);
+    }
+
+    public abstract int CalculateCost();
+
+    /// <summary>
+    ///   Combines two actions to one
+    /// </summary>
+    /// <param name="other">The action this should be combined with. Guaranteed to be combinable</param>
+    /// <returns>Returns the combined action</returns>
+    protected virtual MicrobeEditorActionData CombineGuaranteed(MicrobeEditorActionData other)
+    {
+        throw new NotImplementedException();
+    }
```

Comment:
This could be just be marked abstract so then all child classes are required to implement this.

Reply:
Correct, this can be abstract now. In earlier iterations barley any classes had to implement this, therefore I didn't want to make it a requirement, but now most classes implement this